internal/component/pyroscope/write/write.go

```diff
@@ -466,3 +476,36 @@ func ensureNameMatchesService(lbls labels.Labels) labels.Labels {
 	}
 	return lbls
 }
+
+// validateLabels checks for valid labels and doesn't contain duplicates.
+func validateLabels(lbls labels.Labels) error {
+	if lbls.Len() == 0 {
+		return labelset.ErrServiceNameIsRequired
+	}
+
+	seen := make(map[string]struct{}, lbls.Len())
+
+	for _, l := range lbls {
+		if _, exists := seen[l.Name]; exists {
+			return fmt.Errorf("duplicate label name: %s", l.Name)
+		}
+		seen[l.Name] = struct{}{}
+
+		// Skip validation for reserved labels
+		if l.Name == pyroscope.LabelName {
```

Comment:
Not too sure I understand the reason for skipping.

I think it might make sense to still check if the label value is correct in that case.

Reply:
I've skipped it because it's a `reservedLabelNames ` in the pyroscope `ValidateLabelName` func. However, I can add switch the order and check label values always and then check for valid keys.

Reply:
Updated in 4879713eec22cce3535dc2ea7f59a778f98f5836